File: SWEA/D2/2001. 파리 퇴치/파리 퇴치.java
import java.util.Scanner;

class Solution {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        int t = sc.nextInt();

        for(int test = 0; test < t; test++){
            int n = sc.nextInt();
            int m = sc.nextInt();

            int arr[][] = new int[n][n];

            for(int i = 0; i < n; i++){
                for(int j = 0; j < n; j++){
                    arr[i][j] = sc.nextInt();
                }
            }
            
            int ans = 0;
            for(int i = 0; i <= n - m; i++){
                for(int j = 0; j <= n - m; j++){
                    int sum = 0;
                    for(int x = 0; x < m; x++){
                        for(int y = 0; y < m; y++){
                            sum += arr[i + x][j + y];
                        }
                    }
                    ans = Math.max(ans, sum);
                }
            }

            System.out.printf("#%d %d\n",test + 1,ans);
        }
    }
}

File: SWEA/D2/2001. 파리 퇴치/파리 퇴치.py
t = int(input())

for testcase in range(t):
    n,m = map(int, input().split())

    arr = []
    for _ in range(n):
        arr.append(list(map(int, input().split())))

    answer = 0
    for i in range(n - m + 1):
        for j in range(n - m + 1):
            sum = 0
            for a in range(m):
                for b in range(m):
                    sum +=  arr[a + i][b + j]
            answer = max(answer, sum)

    print(f'#{testcase + 1} {answer}')